synapse/storage/schema/main/delta/74/05_per_user_experimental_features.sql.sqlite

```diff
@@ -0,0 +1,26 @@
+/* Copyright 2023 The Matrix.org Foundation C.I.C
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+-- Table containing experimental features and whether they are enabled for a given user
+CREATE TABLE per_user_experimental_features (
+    -- The User ID to check/set the feature for
+    user_id TEXT NOT NULL,
+    -- Contains features to be enabled/disabled
+    feature TEXT NOT NULL,
+    -- whether the feature is enabled/disabled for a given user, defaults to disabled
+    enabled BOOLEAN DEFAULT 0,
```

Comment:
You no longer need to do this per-DB engine!! :)

> Synapse now requires a SQLite version of 3.27.0 or higher if SQLite is configured
> 
> — https://matrix-org.github.io/synapse/latest/upgrade.html?highlight=sqlite#sqlite-version-requirement

TRUE and FALSE introduced in 3.23.0 (see item 2 on https://www.sqlite.org/releaselog/3_23_0.html)

Unless it doesn't work for some reason I've overlooked..

Reply:
Interesting I got the idea to use the sqlite table from here: https://matrix-org.github.io/synapse/latest/development/database_schema.html#boolean-columns, but it seems that it's not relevant anymore based on your comment. I'll open a PR to fix that as well.

Reply:
oo I see, thanks for updating that